src/rescraper.js

```diff
@@ -41,6 +41,14 @@ export default class Rescraper {
         this.room = room;
     }
 
+    /**
+     * @summary convenience method for updating Announcement
+     * @param {Announcement} announcement announcement instance
+     */
+    setAnnouncement(announcement) {
+        this.announcement = announcement;
```

Comment:
both `setAnnouncement` and `setRescraper` are semi-temporary - a proper solution to the problem would be further decoupling announcements and rescraping to the point. Unfortunately, they naturally ended up a bit too intertwined, so one can't instantiate a `Rescraper` without `Announcement` being instantiated and vice versa...